proxy/envoy/config.go

```diff
@@ -158,62 +162,88 @@ func buildSidecar(env proxy.Environment, sidecar proxy.Node) (Listeners, Cluster
+					m.Name, m.Address, l.Name, l.Address)
+				continue
+			}
+			listeners = append(listeners, m)
+			clusters = append(clusters, c)
+		}
+
+		// set bind to port values for port redirection
+		for _, listener := range listeners {
+			listener.BindToPort = false
 		}
-		listeners = append(listeners, m)
-		clusters = append(clusters, c)
-	}
 
-	// set bind to port values for port redirection
-	for _, listener := range listeners {
-		listener.BindToPort = false
+		// add an extra listener that binds to the port that is the recipient of the iptables redirect
+		listeners = append(listeners, &Listener{
+			Name:           VirtualListenerName,
+			Address:        fmt.Sprintf("tcp://%s:%d", WildcardAddress, env.Mesh.ProxyListenPort),
+			BindToPort:     true,
+			UseOriginalDst: true,
+			Filters:        make([]*NetworkFilter, 0),
+		})
 	}
 
-	// add an extra listener that binds to the port that is the recipient of the iptables redirect
-	listeners = append(listeners, &Listener{
-		Name:           VirtualListenerName,
-		Address:        fmt.Sprintf("tcp://%s:%d", WildcardAddress, env.Mesh.ProxyListenPort),
-		BindToPort:     true,
-		UseOriginalDst: true,
-		Filters:        make([]*NetworkFilter, 0),
-	})
+	// enable HTTP PROXY port if necessary; this will add an RDS route for this port
+	if env.Mesh.ProxyHttpPort > 0 {
+		// only HTTP outbound clusters are needed
+		httpOutbound := buildOutboundHTTPRoutes(env.Mesh, sidecar, instances, services, env.IstioConfigStore)
+		clusters = append(clusters,
+			httpOutbound.clusters()...)
+		listeners = append(listeners,
+			buildHTTPListener(env.Mesh, sidecar, nil, WildcardAddress, int(env.Mesh.ProxyHttpPort), true, false))
+	}
 
 	return listeners.normalize(), clusters.normalize()
 }
 
 // buildRDSRoutes supplies RDS-enabled HTTP routes
 // TODO: this can be optimized by querying for a specific HTTP port in the table
-func buildRDSRoutes(mesh *proxyconfig.ProxyMeshConfig, role proxy.Node,
-	discovery model.ServiceDiscovery, config model.IstioConfigStore) HTTPRouteConfigs {
+func buildRDSRoute(mesh *proxyconfig.ProxyMeshConfig, role proxy.Node, routeName string,
+	discovery model.ServiceDiscovery, config model.IstioConfigStore) *HTTPRouteConfig {
+	port, err := strconv.Atoi(routeName)
+	if err != nil {
+		return nil
+	}
+
 	switch role.Type {
 	case proxy.Ingress:
 		httpRouteConfigs, _ := buildIngressRoutes(mesh, discovery, config)
-		return httpRouteConfigs
+		return httpRouteConfigs[port]
 
 	case proxy.Egress:
-		return buildEgressRoutes(mesh, discovery)
+		return buildEgressRoutes(mesh, discovery)[port]
 
 	case proxy.Sidecar:
 		instances := discovery.HostInstances(map[string]bool{role.IPAddress: true})
 		services := discovery.Services()
-		return buildOutboundHTTPRoutes(mesh, role, instances, services, config)
+		routes := buildOutboundHTTPRoutes(mesh, role, instances, services, config)
+		if port == int(mesh.ProxyHttpPort) {
```

Comment:
I think it would much cleaner to use "all" or "proxy" as the name of the "all routes" - there is no benefit on forcing the route name to be a number, and matching it against the http proxy port ( which in the proposed change can be configured). 

This way it is easy to test and clear that we get the proxy routes.